syntax = "proto3";
package qmtl;

enum WorkType {
    STRATEGY_EXECUTION = 0;
    OTHER = 1;
}

enum WorkStatus {
    PENDING = 0;
    RUNNING = 1;
    COMPLETED = 2;
    FAILED = 3;
}

message WorkItem {
    string id = 1;
    WorkType type = 2;
    bytes payload = 3;
    WorkStatus status = 4;
}
